### PR TITLE
Add additional functions

### PR DESCRIPTION
**Pull request by mwielbut**
- Marshal and unmarshal to JSON
- Save and read from DB (support scan and value interfaces)
- Support AddYears and AddMonths
- Add go.mod

### CODE REVIEW

File: civil.go [hunk 1]
```diff
@@ -121,6 +137,65 @@ func (d *Date) UnmarshalText(data []byte) error {
 	return err
 }
 
+// UnmarshalJSON implements encoding/json Unmarshaler interface
+func (d *Date) UnmarshalJSON(data []byte) error {
```
Comment (bdimcheff) [inline civil.go:142]: did you steal these from stdlib time?

File: civil.go [hunk 2]
```diff
@@ -43,9 +45,11 @@ func DateOf(t time.Time) Date {
 	return d
 }
 
+const rfc3339Date = "2006-01-02"
```
Comment (bdimcheff) [inline civil.go]: these consts are exported in stdlib time, it's probably useful to do it here for other users so they can use them for formatting strings etc.

File: civil.go (hunk 1 above)
Comment (mwielbut) [reply civil.go:142]: I borrowed from our pkg/Date

File: civil.go (hunk 2 above)
Comment (mwielbut) [reply civil.go]: Good point! Fixed.

**Pull request merged**